diff --git a/dist/cjs/server/index.js b/dist/cjs/server/index.js
index 3c7f48e321618692afdfc3aa31375e3758660687..727a7f4ce12ddafdeba131049475a2ff410059e4 100644
--- a/dist/cjs/server/index.js
+++ b/dist/cjs/server/index.js
@@ -601,7 +601,8 @@ var x402ResourceServer = class {
         payTo: resolvedPayTo,
         price: resolvedPrice,
         network: option.network,
-        maxTimeoutSeconds: option.maxTimeoutSeconds
+        maxTimeoutSeconds: option.maxTimeoutSeconds,
+        extra: option.extra
       };
       const requirements = await this.buildPaymentRequirements(resourceConfig);
       allRequirements.push(...requirements);
diff --git a/dist/esm/server/index.mjs b/dist/esm/server/index.mjs
index 5f852152a33d2e86bbb2f8b42ebcd17a5f96c117..3d7171acef928840d6ea3fd15ce0b3a8c6b4df80 100644
--- a/dist/esm/server/index.mjs
+++ b/dist/esm/server/index.mjs
@@ -336,7 +336,8 @@ var x402ResourceServer = class {
         payTo: resolvedPayTo,
         price: resolvedPrice,
         network: option.network,
-        maxTimeoutSeconds: option.maxTimeoutSeconds
+        maxTimeoutSeconds: option.maxTimeoutSeconds,
+        extra: option.extra
       };
       const requirements = await this.buildPaymentRequirements(resourceConfig);
       allRequirements.push(...requirements);
